src/utils/useDimensions.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import { Dimensions, ScaledSize } from 'react-native';
+
+export function useDimensions(): { window: ScaledSize; screen: ScaledSize } {
+  const [dimensions, setDimensions] = useState({
```

Comment:
Can we just listen to `window` changes since we are not using `screen`. So we can also simplify `useDimensions().window;` to `useDimensions();`